### PR TITLE
Reversing the default for ReservedKeywords

### LINKED ISSUE #1385: Add reserved words

### PR DESCRIPTION
**Pull request by gregli-msft**
In general, new features shouldn't change default behavior, they should be an opt-in.  However in this case it makes sense to make it an opt-out, controlled by an internal flag that Cnavs can set.  The nature of this feature and that it is set in ParserOptions and not directly effected by engine.Features (as there may not be one) led us to this conclusion.  

Fixes #1385

### CODE REVIEW

**Comment by MikeStall** [comment]
This makes sense because we are intentionally changing the default behavior.  So we want to get in for GA

File: src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs [hunk 1]
```diff
@@ -26,6 +26,12 @@ internal class InternalSetup
         /// </summary>
         internal bool DisableMemoryChecks { get; set; }
 
+        /// <summary>
+        /// By default, we run the result back through the serializer and deserialize again to se if we get the same result.
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs:30]: >se [](http://example.com/codeflow?start=95&length=2)

nit: typo

File: src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs [hunk 2]
```diff
@@ -26,6 +26,12 @@ internal class InternalSetup
         /// </summary>
         internal bool DisableMemoryChecks { get; set; }
 
+        /// <summary>
+        /// By default, we run the result back through the serializer and deserialize again to se if we get the same result.
+        /// For some tests, for example reserved words, the deserialization will not work properly.
+        /// </summary>
+        internal bool SkipDeserializeComparison { get; set; }
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs:33]: >SkipDeserializeComparison [](http://example.com/codeflow?start=22&length=25)

Do we need the new flag - can we just infer this if the test case has an error that we shouldn't try serializing?

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs (hunk 2 above)
Comment (MikeStall) [reply src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs:33]: ?

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs [hunk 3]
```diff
@@ -362,12 +367,12 @@ protected override async Task<RunResult> RunAsyncInternal(string expr, string se
...
                     {
                         // Serialization test. Serialized expression must produce an identical result.
-                        ParserOptions options = new ParserOptions() { NumberIsFloat = true };
+                        options.NumberIsFloat = true;
                         newValueDeserialized = await engine.EvalAsync(sb.ToString(), CancellationToken.None, options, runtimeConfig: runtimeConfig).ConfigureAwait(false);
                     }
                     else
                     {
-                        throw;
+                        throw e;
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs:375]: > throw e; [](http://example.com/codeflow?start=23&length=9)

nit - changing to 'throw e' means we lose the callstack of the original exception. 
Can we keep as 'throw' ?

**Review by MikeStall — DISMISSED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs (hunk 2 above)
Comment (MikeStall) [reply src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs:33]: Tracking #1409 to remove this flag.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs (hunk 2 above)
Comment (gregli-msft) [reply src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/InternalSetup.cs:33]: Flag removed and tests skipped for now.  I'll make a note in 1409, but the tests should be unskipped when that is done.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs (hunk 3 above)
Comment (gregli-msft) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs:375]: Fixed.

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**